core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -436,12 +436,24 @@ oppia.directive('conversationSkin', [
             }
 
             if (ExplorationPlayerStateService.isStateTerminal(stateName)) {
-              ExplorationRecommendationsService.getRecommendedSummaryDicts(
-                ExplorationPlayerStateService.getAuthorRecommendedExpIds(
-                  stateName),
-                function(summaries) {
-                  $scope.recommendedExplorationSummaries = summaries;
-                });
+              $scope.parentExplorationIds = UrlService.getParamValuesAsList(
+                'parent');
+              if ($scope.parentExplorationIds) {
+                var parentExplorationId = $scope.parentExplorationIds[
+                  $scope.parentExplorationIds.length - 1];
+                ExplorationRecommendationsService.getRecommendedSummaryDicts(
```

Comment:
Always try to reduce duplication between if/else branches. At the outset, above the $scope.parentExplorationIds, declare `var recommendedExplorationIds = [];`. Then populate recommendedExplorationIds in each of the if/else branches. Then finally, outside the if/else branch, do:

```
                ExplorationRecommendationsService.getRecommendedSummaryDicts(
                  recommendedExplorationIds,
                  function(...) {
                    ...
                  });
```

Reply:
Done